torch/onnx/__init__.py

```diff
@@ -388,8 +388,18 @@ def forward(self, x):
             fallback=fallback,
         )
     else:
+        import warnings
+
         from torch.onnx.utils import export
 
+        warnings.warn(
+            "The legacy ONNX export logic is deprecated and will be removed in a future release. "
```

Comment:
I think we can mention PyTorch 2 and weaken the “will be removed” clause. I guess the idea is to encourage users to try it out and report issues + being informative on depreciation

Reply:
Also motivate users to turn on dynamo=Ture primarily because it is better, not for the fear of the old logic being removed.

Reply:
How about

```python
        warnings.warn(
            "You are using the legacy ONNX export logic. Starting with PyTorch 2.8, we recommend using the new export "
            "logic by setting `dynamo=True` in the `torch.onnx.export` API. The new logic offers better performance, "
            "support for modern features, and improved debugging capabilities. Please try it out and report any issues "
            "you encounter to help us improve.",
            category=DeprecationWarning,
            stacklevel=3,
        )
```

Reply:
That sounds good to me!